config/plugins/interactive_environments/askomics/templates/askomics.mako

```diff
@@ -0,0 +1,50 @@
+<%namespace name="ie" file="ie.mako" />
+<%
+import random
+import string
+# Sets ID and sets up a lot of other variables
+ie_request.load_deploy_config()
+
+# Get a random API key
+askomics_api_key = ''.join([random.choice(string.ascii_letters + string.digits) for n in xrange(20)])
```

Comment:
there's a built-in, randomly generated password available to you already via `ie_request.notebook_pw`

e.g. in use here.
https://github.com/galaxyproject/galaxy/blob/dev/config/plugins/interactive_environments/jupyter/templates/jupyter.mako#L15

Reply:
fixed :-)